tests/e2e/env/bin/docker-compose.js

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env node
```

Comment:
Is there some way that we can remove the need for this script and invert the `docker-compose` dependency altogether? I just feel like we shouldn't be so opinionated if we can avoid it. As far as I can tell, the only reasons we need it is to create the docker-compose command with the configuration overrides, as well as copy the initialization scripts into appropriate directories so that they can be included.

For the first problem, since we're already using `npm explore`, could we just have them pass the path to their `docker-compose.yml` in the command? Something like `npm explore @woocommerce/e2e-environment -- npm run docker:up —f path/to/file`. This also allows for a different name that will prevent any conflicts too.

For the second problem, could we make use of DockerFile args or environment variables?

Reply:
> could we just have them pass the path to their docker-compose.yml in the command?

One of the goals of the component is to provide a standard/common test environment that can be used across all our WC projects/repos. Moving most of the configuration to the package consumer's environment is counterproductive to that.

> I just feel like we shouldn't be so opinionated if we can avoid it.

Quite a bit of the configuration can be overridden in the consumer project if necessary. I removed the "local" configuration files if they were redundant to the default provided in the component.

Long term I'd like the environment to be as flexible as possible while maintaining minimal configuration in the consuming project.

Reply:
Just brainstorming here, as we've established small PRs can be used to iterate!

> Moving most of the configuration to the package consumer's environment is counterproductive to that.

I was more thinking just in the context of their custom configurations. We could likely support both the default (via `npm explore`) and an override that has them pass the configuration file location. In the default case they would change absolutely nothing, but in the custom case, they would use a slightly different command.

Specifically I'm just trying to find ways to remove all of the hard-coded paths and move the docker-compose.yml configuration out of the project root, so that different consumers of this package can use it however they see fit. That said, I think we could also accomplish something similar by adding more environment variables or arguments, but that's personal preference I think.

> Long term I'd like the environment to be as flexible as possible while maintaining minimal required configuration in the consuming project.

Agreed. I'll think on this one and maybe draft an experimental PR to see what that may look like once we've merged this one. Either way, great work here!

Reply:
> Specifically I'm just trying to find ways to remove all of the hard-coded paths

I think we can iterate on that.

> move the docker-compose.yml configuration out of the project root

The file currently in the root specifies where it is in relation to the one in the component. The component isn't aware of the location of the one in the root.